chainer/training/extensions/evaluator.py

```diff
@@ -236,3 +265,33 @@ def finalize(self):
         """
         for iterator in six.itervalues(self._iterators):
             iterator.finalize()
+
+
+class _IteratorProgressBar(util.ProgressBar):
```

Comment:
This class requires some attributes in `iterator`.
How about checking them in `__init__`?
Namely
* `hasattr(iterator, 'current_position')`
* `hasattr(iterator, 'epoch_detail')`

Reply:
I tried not to use `__init__` to check attributes: e96267533ecd49bef6e19b367bc47983995a46c4
This implementation is dependent on the discussion here: https://github.com/chainer/chainer/pull/6474#discussion_r288059939